src/managers/SessionLockManager.cpp

```diff
@@ -13,7 +13,8 @@ SSessionLockSurface::SSessionLockSurface(SP<CSessionLockSurface> surface_) : sur
     listeners.map = surface_->events.map.registerListener([this](std::any data) {
         mapped = true;
 
-        g_pInputManager->simulateMouseMovement();
+        if (!g_pCompositor->m_pLastFocus)
```

Comment:
doing this may prevent the lock surface from being focused

Reply:
The focus is reset when the session is locked here: https://github.com/hyprwm/Hyprland/blob/1fb720b62aeb474873ba43426ddc53afde1e6cdd/src/managers/SessionLockManager.cpp#L88
so `g_pCompositor->m_pLastFocus` should always be null after the session is locked.

But I'm happy to remove the check, it doesn't affect the swaylock issue and just avoids extra unnecessary motion events.

Reply:
Actually nevermind just checked, removing it results in extra motion events if the first surface mapped was the one under the cursor and was thus focused (the subsequent calls to `mouseMoveUnified` will have `notFirstFocus == true` and will therefore produce motion events).